java/maven/src/org/netbeans/modules/maven/layer.xml

```diff
@@ -33,6 +33,22 @@
             <folder name="Maven2">
                 <attr name="displayName" bundlevalue="org.netbeans.modules.maven.newproject.Bundle#Templates/Project/Maven2"/>
                 <attr name="position" intvalue="100"/>
+
+                <!-- Simple Java project structure -->
+                <folder name="JavaApplication2">
+                    <folder name="src">
+                        <folder name="main">
+                            <folder name="java">
+                                <folder name="${packagePath}">
+                                    <file name="${mainClassName}.java" url="nbresloc:/org/netbeans/modules/maven/resources/App.java.template">
```

Comment:
Using new substitution mechanism for names is OK, but the syntax! Let's make the syntax pluggable! Sváťo, what do you think about:

If there is an `<attr name="javax.script.ScriptEngine" stringvalue="freemarker"/>` attribute on a file or folder, apply the specified engine (e.g. `freemarker`) to the name of the file/folder as well.

Reply:
well, respecting the ScriptEngine could be helpful in some more complicated cases, asi the expression can reference context objects for example; but in simple ones like this ...

Reply:
Yes, it is an overkill.

On the other hand, the templating engine must be around and it provides familiar syntax. E.g. one would be using the same syntax for the file name as well as for the file content.